src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1331,14 +1368,17 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
     std::vector<double> EclipseGrid::activeVolume() const {
         std::vector<double> active_volume( this->m_nactive );
 
-        #pragma omp parallel for schedule(static)
-        for (std::size_t active_index = 0; active_index < this->m_active_to_global.size(); active_index++) {
-            std::array<double,8> X;
-            std::array<double,8> Y;
-            std::array<double,8> Z;
-            auto global_index = this->m_active_to_global[active_index];
-            this->getCellCorners(global_index, X, Y, Z );
-            active_volume[active_index] = calculateCellVol(X, Y, Z);
+        {
+            ThreadGuard tg(1);
```

Comment:
Why 1? Doesn't that mean no parallelism in the loop?

Reply:
Hmmm - I had understood it differently, but you might be right--will have to check again.

Reply:
He is right, opm_get_num_threads return currently number of active threads - which means 1 outside a parallel section.

Reply:
That was also my conclusion - ;-(  I'll do some DuckDuckGo - but I guess I am quite close to just dropping all thread manipulations. Damn difficult ...